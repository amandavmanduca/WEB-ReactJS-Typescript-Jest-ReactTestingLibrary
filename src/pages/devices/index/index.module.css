
.deviceCardBox {
    width: 100%;
    display: grid;
    grid-template-columns: 1fr 1fr 1fr;
    gap: 20px;
}

@media (max-width: 1000px) {
    .deviceCardBox {
        grid-template-columns: 1fr 1fr;
    }
}

@media (max-width: 700px) {
    .deviceCardBox {
        grid-template-columns: 1fr;
    }
}

.flexBox {
    display: flex;
    flex-direction: column;
    gap: 10px;
}

.noDataFoundBox {
    width: 100%;
    margin-top: 50px;
    height: 100%;
    gap: 10px;
    display: flex;
    justify-content: center;
    align-items: center;
    flex-direction: column;
}