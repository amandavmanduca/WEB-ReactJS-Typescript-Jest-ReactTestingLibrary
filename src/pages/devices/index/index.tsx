import { useEffect } from 'react';
import { Link, useNavigate } from 'react-router-dom';
import { Button } from '../../../atomic/atom/Button';
import Text from '../../../atomic/atom/Text';
import { Device } from '../../../features/common/types';
import { DeviceCard } from '../../../features/devices/components/DeviceCard';
import { useDeleteDevice } from '../../../features/devices/hooks/useDeleteDevice';
import { useListDevices } from '../../../features/devices/hooks/useListDevices';
import styles from './index.module.css';


const DevicesPage = () => {
    const { listDevices, data } = useListDevices()
    const { deleteDevice } = useDeleteDevice()
    const navigate = useNavigate();

    useEffect(() => {
        listDevices()
    // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [])

    async function handleDelete(id: string) {
        if (id) {
            await deleteDevice(id)
            listDevices()
        }
    }

    return (
        <div className={styles.flexBox}>
            <div className={styles.flexBox}>
                <div>
                    <Button
                        onClick={() => navigate('/device/create')}
                    >
                        Criar dispositivo
                    </Button>
                </div>
                <Text type="h1">Listagem de dispositivos</Text>
            </div>
            {data?.length > 0 ? (
                <div data-testid="device-card-section" className={styles.deviceCardBox}>
                    {data?.map((device: Device) => (
                        <DeviceCard
                            key={device.id}
                            device={device}
                            deleteDevice={() => handleDelete(device.id)}
                        />
                    ))}
                </div>
            ) : (
                <div data-testid="empty-data" className={styles.noDataFoundBox}>
                    <Text type="h1">Nenhum dispositivo encontrado</Text>
                    <Link to="/device/create"><Text type="p">Comece a cadastrar clicando aqui.</Text></Link>
                    <img src="/icons/empty-folder.png" alt="Empty" />
                </div>
            )}
        </div>
    )
}

export default DevicesPage