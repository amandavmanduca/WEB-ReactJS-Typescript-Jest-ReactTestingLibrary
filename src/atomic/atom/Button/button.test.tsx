import { render, screen, cleanup, fireEvent } from '@testing-library/react';
import { Button } from './index';
import renderer from 'react-test-renderer';

afterEach(() => cleanup())

test('Should render button component', async () => {
  const handleClick = jest.fn()
  render(<Button onClick={handleClick}>Test Button</Button>)
  const button = screen.getByTestId('button')

  expect(button).toBeInTheDocument()
  expect(button).toBeEnabled()
  expect(button).toHaveTextContent('Test Button')
});

test('Should call function when clicking button component', async () => {
  const handleClick = jest.fn()
  render(<Button onClick={handleClick}>Test Button</Button>)
  const button = screen.getByTestId('button')
  fireEvent.click(button)
  expect(button).toBeEnabled()
  expect(handleClick).toBeCalled()
});

test('Should render button component as disabled', async () => {
  const handleClick = jest.fn()
  render(<Button onClick={handleClick} disabled={true}>Test Button</Button>)
  const button = screen.getByTestId('button')
  expect(button).toBeDisabled()
});

test('Button component should matches snapshot', async () => {
    const tree = renderer.create(<Button onClick={() => console.log('clicking button')}>Test Button</Button>).toJSON()
    expect(tree).toMatchSnapshot()
})
