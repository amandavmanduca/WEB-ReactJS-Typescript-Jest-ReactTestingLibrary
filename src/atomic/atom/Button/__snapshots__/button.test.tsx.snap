// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should matches snapshot 1`] = `
<button
  className="button"
  data-testid="button"
  disabled={false}
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "#fc8eac",
    }
  }
>
  <p
    className="buttonText"
  >
    Test Button
  </p>
</button>
`;
